drivers/gpio/gpio_tle9104.c

```diff
@@ -0,0 +1,544 @@
+			return -ENODEV;
+		}
+
+		result = gpio_pin_configure_dt(&config->gpio_enable, GPIO_OUTPUT_ACTIVE);
+		if (result != 0) {
+			LOG_ERR("failed to enable TLE9104");
+			return result;
+		}
+	}
+
+	result = gpio_pin_configure_dt(&config->gpio_reset, GPIO_OUTPUT_ACTIVE);
+	if (result != 0) {
+		LOG_ERR("failed to initialize GPIO for reset");
+		return result;
+	}
+
+	k_busy_wait(TLE9104_RESET_DURATION_TIME_IN_US);
+	gpio_pin_set_dt(&config->gpio_reset, 0);
+	k_busy_wait(TLE9104_RESET_DURATION_WAIT_TIME_IN_US +
+		    TLE9104_RESET_DURATION_WAIT_TIME_SAFETY_MARGIN_IN_US);
+
+	/*
+	 * The first read value should be the ICVID, this acts also as the setup of the
+	 * global status register address.
+	 */
+	result = tle9104_transceive_frame(dev, false, TLE9104REGISTER_GLOBALSTATUS, 0x00, &read_reg,
+					  &register_icvid);
+	if (result != 0) {
+		return result;
+	}
+
+	if (read_reg != TLE9104REGISTER_ICVID) {
+		LOG_ERR("expected to read register ICVID, got instead 0x%02X", read_reg);
+		return -EIO;
+	}
+
+	if (register_icvid != TLE9104_ICVERSIONID) {
+		LOG_ERR("got unexpected IC version id 0x%02X", register_icvid);
+		return -EIO;
+	}
+
+	result = tle9104_transceive_frame(dev, false, TLE9104REGISTER_GLOBALSTATUS, 0x00, &read_reg,
+					  &register_globalstatus);
+	if (result != 0) {
+		return result;
+	}
+
+	if (read_reg != TLE9104REGISTER_GLOBALSTATUS) {
+		LOG_ERR("expected to read register GLOBALSTATUS, got instead 0x%02X", read_reg);
+		return -EIO;
+	}
+
+	if ((register_globalstatus & TLE9104_GLOBALSTATUS_POR_LATCH) == 0) {
+		LOG_ERR("no power on reset detected");
+		return -EIO;
+	}
+
+	result = tle9104_write_register(dev, TLE9104REGISTER_CFG, register_cfg);
+	if (result != 0) {
+		LOG_ERR("unable to write configuration");
+		return result;
+	}
+
+	register_globalstatus = 0x00;
+	/* disable communication watchdog */
+	tle9104_set_cfg_cwdtime(&register_cfg, 0);
+	/* enable outputs */
+	register_globalstatus |= TLE9104_GLOBALSTATUS_OUTEN_BIT;
+
+	result = tle9104_write_register(dev, TLE9104REGISTER_GLOBALSTATUS, register_globalstatus);
+	if (result != 0) {
+		LOG_ERR("unable to write global status");
+		return result;
+	}
+
+	return 0;
+}
+
+BUILD_ASSERT(CONFIG_GPIO_TLE9104_INIT_PRIORITY > CONFIG_SPI_INIT_PRIORITY,
+	     "TLE9104 must be initialized after SPI");
+
+#define TLE9104_INIT_GPIO_FIELDS(inst, gpio)                                                       \
+	COND_CODE_1(DT_INST_NODE_HAS_PROP(inst, gpio),                                             \
+		    (GPIO_DT_SPEC_GET_BY_IDX(DT_DRV_INST(inst), gpio, 0)), ({0}))
+
+#define TLE9104_INIT(inst)                                                                         \
+	static const struct tle9104_config tle9104_##inst##_config = {                             \
+		.common =                                                                          \
+			{                                                                          \
```

Comment:
the curly brace should be inline for these (I know clang-format is messing these up, sorry, no idea how to fix that)

Reply:
Do you mean it breaks https://gitlab.com/MISRA/MISRA-C/MISRA-C-2012/Example-Suite/-/blob/master/R_09_02.c of https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html?

And what would be the correct way, maybe this?

		.common = {                                                                         \
				.port_pin_mask = GPIO_PORT_PIN_MASK_FROM_DT_INST(inst),            \
			},                                                                         \